server.go

```diff
@@ -149,6 +150,30 @@ func CustomCodec(codec Codec) ServerOption {
 	}
 }
 
+// CustomCodecs returns a ServerOption that sets the codecs that can be used based on the
+// content-type subtype, ie if a codec returns "proto" from calling String(), this will
+// be used if the content-type is "application/grpc+proto". If there is no subtype
+// on the content-type, the codec set by CustomCodec will be used, or if CustomCodec
+// is not set, then the default codec will be used.
+func CustomCodecs(codecs ...Codec) ServerOption {
+	m := make(map[string]Codec, len(codecs))
+	for _, codec := range codecs {
+		// this is a one-time server setup operation
+		// note that http/2 transmits headers as lowercase
+		name := strings.TrimSpace(strings.ToLower(codec.String()))
```

Comment:
Still don't understand - trying to make it so that codec can be not used effectively (and just have a passthrough) - `codec.String()` is effectively used as subtype right now as it is, so I wouldn't want to add another function that also does this.

The goal is to give the ability for the client to specify the subtype without relying on `codec.String()`, ie not relying on the codec at all, which is needed when the encoding and transport are decoupled (grpc-go relies on the marshaling/unmarshaling being done internally using just the `Codec` interface).